Mathlib/Topology/Order/LowerUpperTopology.lean

```diff
@@ -247,6 +347,86 @@ end PartialOrder
 
 end LowerTopology
 
+
+namespace UpperTopology
+
+/-- The complements of the lower closures of finite sets are a collection of upper sets
+which form a basis for the upper topology. -/
+def upperBasis (α : Type _) [Preorder α] :=
+  { s : Set α | ∃ t : Set α, t.Finite ∧ (lowerClosure t : Set α)ᶜ = s }
+
+section Preorder
+
+variable (α)
+variable [Preorder α] [TopologicalSpace α] [UpperTopology α] {s : Set α}
+
+lemma topology_eq : ‹_› = generateFrom { s | ∃ a : α, (Iic a)ᶜ = s } := topology_eq_upperTopology
+
+variable {α}
+
+/-- If `α` is equipped with the upper topology, then it is homeomorphic to `WithUpperTopology α`.
+-/
+def withUpperTopologyHomeomorph : WithUpperTopology α ≃ₜ α :=
+  WithUpperTopology.ofUpper.toHomeomorphOfInducing ⟨by erw [topology_eq α, induced_id]; rfl⟩
+
+theorem isOpen_iff_generate_Iic_compl : IsOpen s ↔ GenerateOpen { t | ∃ a, (Iic a)ᶜ = t } s := by
+  rw [topology_eq α]; rfl
+
+instance  [Preorder α] [TopologicalSpace α] [UpperTopology α] : LowerTopology (αᵒᵈ) where
+  topology_eq_lowerTopology := by
+    refine topologicalSpace_eq ?_
+    rw [(UpperTopology.topology_eq (α))]
```

Comment:
```suggestion
instance [Preorder α] [TopologicalSpace α] [UpperTopology α] : LowerTopology (αᵒᵈ) where
  topology_eq_lowerTopology := topology_eq_upperTopology (α := α)
```
A bit more of defeq abuse, but the whole "apply the lemmas to the dual and let Lean figure out it works" is defeq abuse already.